antarest/study/business/link_management.py

```diff
@@ -14,30 +14,152 @@
+
+
+LinkInfoDTOType = t.Union[LinkInfoDTO820, LinkInfoDTOBase]
+
+
+class LinkInfoFactory:
+    @staticmethod
+    def create_link_info(version: int, **kwargs) -> LinkInfoDTOType:
+        """
+        Creates a LinkInfoDTO object corresponding to the specified version.
+
+        Args:
+            version (int): The study version.
+            kwargs: The arguments passed for the DTO creation.
+
+        Returns:
+            LinkInfoDTOType: An object of type LinkInfoDTOBase or LinkInfoDTO820 depending on the version.
+        """
+        LinkInfoFactory._check_version_coherence(version, **kwargs)
+        link_info = LinkInfoFactory._initialize_link_info(version, **kwargs)
+        LinkInfoFactory._set_default_filters(version, link_info)
+
+        return link_info
+
+    @staticmethod
+    def _initialize_link_info(version: int, **kwargs) -> LinkInfoDTOType:
+        """
+        Initializes the LinkInfoDTO object based on the study version.
+
+        Args:
+            version (int): The study version.
+            kwargs: The arguments passed for the DTO creation.
+
+        Returns:
+            LinkInfoDTOType: An object of type LinkInfoDTOBase or LinkInfoDTO820 depending on the version.
+        """
+        if version >= 820:
+            return LinkInfoDTO820(**kwargs)
+        return LinkInfoDTOBase(**kwargs)
+
+    @staticmethod
+    def _set_default_filters(version: int, link_info: LinkInfoDTOType) -> None:
+        """
+        Sets default filters if the study version is 820 or higher.
+
+        Args:
+            version (int): The study version.
+            link_info (LinkInfoDTOType): The created DTO object.
+        """
+        if version >= 820 and isinstance(link_info, LinkInfoDTO820):
+            if link_info.filter_synthesis is None:
+                link_info.filter_synthesis = FilteringOptions.FILTER_SYNTHESIS
+            if link_info.filter_year_by_year is None:
+                link_info.filter_year_by_year = FilteringOptions.FILTER_YEAR_BY_YEAR
+
+    @staticmethod
+    def _check_version_coherence(version: int, **kwargs) -> None:
+        """
+        Checks if filters are provided for a study version lower than 820.
+
+        Args:
+            version (int): The study version.
+            kwargs: The arguments passed for the DTO creation.
+
+        Raises:
+            InvalidFieldForVersionError: If filters are provided for a version lower than 820.
+        """
+        filters_provided = kwargs.get("filter_synthesis") is not None or kwargs.get("filter_year_by_year") is not None
+        if version < 820 and filters_provided:
+            raise InvalidFieldForVersionError(
+                "Filters filter_synthesis and filter_year_by_year cannot be used for study versions lower than 820."
+            )
+
+    @staticmethod
+    def create_parameters(
+        study_version: int, link_creation_info: LinkInfoDTOType
+    ) -> t.Dict[str, t.Union[str, bool, float]]:
```

Comment:
As filter_synthesis and filter_year_by_year are optional = None you should add None inside the typing Union to fix the mypy issue.
Also change it inside the little following doc